### DISCUSSION TITLE
怎么支持android x86 abi？

### DISCUSSION THREAD

**Discussion by Tan-Tshunglee** [Tencent/libpag#482] [Q&A] [answered]
官方推到maven center的android包里面支持arm 相关的abi支持，贵司有计划新增x86 abi的支持不？如果需要自己编译的话，有文档可以瞅瞅的不？

**Comment by Tan-Tshunglee** [comment]
在windows上用msys2 配合 android ndk on windows 把libffav 编译出来之后，再尝试编译libpag时，发现这个是个大坑

**Reply by domchen** [reply, marked as answer]
x86架构的设备是可以直接用arm架构的库的。你这里必须专门编译 x86 架构的原因是？如果你必须要编译x86架构，严格跟着ReadMe里的开发指南一节文档步骤走，配置完所有依赖项，然后用Android Studio编译那个android demo工程就行，不用任何命令行操作。目标架构在Android Studio里选一下。本身那个demo项目跑模拟器的时候编译出来的就是x86架构。全过程目前还没听到过有任何问题反馈。推测如果你遇到坑大概率是因为windows的环境问题，换成macOS编译的话目测坑会少90%以上，各种开发和编译环境在windows下都是hard模式。
